app/controllers/nsm/imports_controller.rb

```diff
@@ -63,6 +58,42 @@ def self.model_class
 
     private
 
+    def handle_missing_parameter
+      @form_object = ImportForm.new
+      @form_object.errors.add(:file_upload, :blank)
+      render :new
+    end
+
+    def handle_standard_error(error)
+      @form_object = ImportForm.new
+      @form_object.errors.add(:file_upload, :unexpected_error, message: error.message)
+      render :new
+    end
+
+    def process_successful_import(claim)
+      hash = prepare_claim_hash
+      version = extract_version_from_xml
+
+      import_claim(claim, hash, version)
+
+      redirect_to edit_nsm_steps_claim_type_path(claim.id), flash: { success: build_message(claim) }
+    end
+
+    def prepare_claim_hash
+      hash = Hash.from_xml(xml_file.to_s)['claim']
```

Comment:
Probably cleaner to do this in 1 line by using the [except](https://apidock.com/rails/Hash/except) method: 
`Hash.from_xml(xml_file.to_s)['claim'].except(:version)`